nitrates/analysis_seeds/UpperLimits.py

```diff
@@ -0,0 +1,308 @@
+import numpy as np
+from astropy.io import fits 
+from astropy.table import Table, vstack
+from astropy.wcs import WCS
+import os
+import time
+import matplotlib.pyplot as plt
+from matplotlib import cm
+import pandas as pd
+from copy import copy, deepcopy
+import logging, traceback
+import sys
+import argparse
+import time
+
+from ..lib.event2dpi_funcs import det2dpis, mask_detxy
+from ..lib.sqlite_funcs import get_conn
+from ..lib.dbread_funcs import get_info_tab
+from ..lib.coord_conv_funcs import  convert_theta_phi2radec
+
+from ..analysis_seeds.do_full_rates import *
+
+from ..lib.calc_BAT_ul import *
+
+start_time = time.time()
+
+def cli():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--work_dir", type=str, help="Results directory", default="/gpfs/group/jak51/default/F702399789/")
+    parser.add_argument("--evfname", type=str, help="Event data file", default=None)
+    parser.add_argument("--dmask", type=str, help="Detmask fname", default=None)
+    parser.add_argument(
+        "--att_fname", type=str, help="Fname for that att file", default=None
+    )
+    parser.add_argument(
+        "--trig_time",
+        type=str,
+        help="Time of trigger, in either MET or a datetime string",
+    )
+    parser.add_argument(
+        "--dbfname", type=str, help="Name to save the database to", default=None
+    )
+    parser.add_argument(
+        "--bkg_fname",
+        type=str,
+        help="Name of the file with the bkg fits",
+        default="bkg_estimation.csv",
+    )
+    parser.add_argument(
+        "--twind",
+        type=float,
+        help="Number of seconds to go +/- from the trigtime",
+        default=20 * 1.024,
+    )
+    parser.add_argument(
+        "--min_dt", type=float, help="Min time from trigger to do", default=None
+    )
+    parser.add_argument(
+        "--drm_dir_old",
+        type=str,
+        help="drm_directory",
+        default="/gpfs/group/jak51/default/responses/rsp_maskweight/",
+    )
+    parser.add_argument(
+        "--drm_dir",
+        type=str,
+        help="drm_directory",
+        default="/gpfs/group/jak51/default/responses/rsp_NITRATES/",
+    )
+    parser.add_argument(
+        "--resp_dname",
+        type=str,
+        help="new responses directory",
+        default="/gpfs/group/jak51/default/rsps4limits2/"
+    )
+    parser.add_argument(
+        "--api_token",
+        type=str,
+        help="EchoAPI key for interactions.",
+        default=None
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+
+
+
+
+
+def main(args):
+    logging.basicConfig(
+        filename="upper_limits_analysis.log",
+        level=logging.DEBUG,
+        format="%(asctime)s-" "%(levelname)s- %(message)s",
+    )
+
+    work_dir=args.work_dir
+    resp_dname=args.resp_dname
```

Comment:
Make the default none for resp_dname, if it's still none then import the one from config.py

Reply:
Ive made it such that I import from config.py now